### PR TITLE
feat: add support for go-restful v3

### PR DESCRIPTION
**Pull request by luqmansen**
#967

### CODE REVIEW

**Comment by luqmansen** [comment]
Sorry, forgot to sign the contributor agreement, now I have signed it, how to ask the bot to recheck my PR?

File: module/apmrestfulv3/filter_test.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package apmrestfulv3_test
+
+import (
+	"errors"
+	"go.elastic.co/apm/module/apmrestfulv3"
```
Comment (axw) [inline module/apmrestfulv3/filter_test.go:22]: can you move this down with the other `go.elastic.co/apm` imports please? (`make fmt` should do that for you)

File: module/apmrestfulv3/go.mod [hunk 2]
```diff
@@ -0,0 +1,15 @@
+module go.elastic.co/apm/module/apmrestfulv3
+
+require (
+	github.com/emicklei/go-restful v2.9.6+incompatible
```
Comment (axw) [inline module/apmrestfulv3/go.mod:4]: I don't think the v2.9.6 module is needed?

**Review by axw — COMMENTED** [review]
Thanks @luqmansen! Just a couple of minor issues. Please run `make fmt update-modules scripts/Dockerfile-testing`, then it should be good to merge.

File: module/apmrestfulv3/filter_test.go (hunk 1 above)
Comment (luqmansen) [reply module/apmrestfulv3/filter_test.go:22]: sure

**Comment by luqmansen** [comment]
Hey, I updated the PR, can you please review it again? Thank you @axw

**Review by axw — APPROVED** [review]
LGTM, thanks! I'll just need to get it to run through CI now.

**Comment by axw** [comment]
jenkins run the tests please

**Comment by axw** [comment]
@luqmansen CI looks good except that it fails on older versions of Go that don't support modules. Can you please add a build tag to each file in apmrestfulv3 _except_ for doc.go: `// +build go1.11`. If you're unfamiliar with using build tags, here's an example for reference: https://github.com/elastic/apm-agent-go/blob/51532741429a0baee951e5237c39a1bdf9432376/module/apmgopgv10/hook.go#L18-L20

**Comment by luqmansen** [comment]
Sure, will update this soon

**Comment by axw** [comment]
:tada: thanks again @luqmansen!

**Comment by luqmansen** [comment]
Thank you! @axw

**Pull request merged**